test/rulesets/Base/array.jl

```diff
@@ -14,7 +14,7 @@ end
     A = randn(3, 2)
     B = randn(3, 1)
     C = randn(3, 3)
-    test_rrule(reduce, hcat ⊢ nothing, [A, B, C])
+    test_rrule(reduce, hcat ⊢ DoesNotExist(), [A, B, C])
```

Comment:
`hcat ⊢ DoesNotExist()`
this makes me think that `NO_FIELDS` should be `DoesNotExist()` rather than `Zero()`

Reply:
I was surprised that `NO_FIELDS` was `Zero()` rather than `DoesNotExist()` when I first saw it to be honest.

Reply:
It is a discussion we have had a few times and flipped back and forth.
It's why it is a constant rather than being hard-coded